pybossa/cache/site_stats.py

```diff
@@ -17,19 +17,22 @@
 # along with PYBOSSA.  If not, see <http://www.gnu.org/licenses/>.
 """Cache module for site statistics."""
 from functools import wraps
-from sqlalchemy.sql import text
+
 from flask import current_app
+from sqlalchemy.sql import text
 
-from pybossa.core import db
-from pybossa.cache import cache, memoize, ONE_DAY, ONE_WEEK
 import pybossa.app_settings as app_settings
-from pybossa.cache import sentinel, management_dashboard_stats
+from pybossa.cache import ONE_DAY, ONE_WEEK, \
+    memoize_with_l2_cache, TWO_WEEKS
 from pybossa.cache import get_cache_group_key, delete_cache_group
+from pybossa.cache import sentinel, management_dashboard_stats
+from pybossa.core import db
 
 session = db.slave_session
 
 
-@cache(timeout=ONE_DAY, key_prefix="site_n_auth_users")
+@memoize_with_l2_cache(timeout=ONE_DAY, timeout_l2=TWO_WEEKS,
```

Comment:
From Xi: Cache will still timeout in one day. L2 cache will timeout after 2 weeks. After L1 cache timeout in one day, the first query will fetch data from database. in the meantime if another query comes in, it will obtain data from L2 cache as previous query is already busy(has obtained mutex lock) in obtaining data from database